backend/src/main/repository/PromotionRepository.ts

```diff
@@ -40,9 +40,16 @@ export class PromotionRepository extends Repository<Promotion> {
     }
 
     if (promotionQuery?.cuisine) {
-      queryBuilder.andWhere('promotion.cuisine = :cuisine', {
-        cuisine: promotionQuery.cuisine,
-      });
+      if (Array.isArray(promotionQuery.cuisine)) {
+        // https://github.com/typeorm/typeorm/issues/1239#issuecomment-366955628
+        queryBuilder.andWhere('promotion.cuisine in (:...cuisine)', {
+          cuisine: promotionQuery.cuisine,
+        });
+      } else {
+        queryBuilder.andWhere('promotion.cuisine = :cuisine', {
+          cuisine: promotionQuery.cuisine,
+        });
+      }
```

Comment:
Pretty minor optimization, I think you can assign `promotionQuery.cuisine` to a variable before this if-else block and make use of it throughout cause it looks like there are 3 usages

Reply:
Oh, do you mind explaining how it optimizes the performance? Or do you mean like for readability